Sources/KlaviyoCore/Networking/PrivateMethods.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  File.swift
+//
+//
+//  Created by Ajay Subramanya on 8/29/24.
+//
+
+import Foundation
+
+@_spi(KlaviyoPrivate)
+/// Used to override SDK defailts for INTERNAL USE ONLY
+/// - Parameter url: The  URL to use for Klaviyo client APIs, This is used internally to test the SDK against different backends, DO NOT use this in your apps.
+/// - Parameter name: The name of the SDK, defaults to swift but react native will pass it's own name. DO NOT override this in your apps as our backend will not accept unsupported values here and your network requests will fail.
+/// - Parameter version: The version of the swift SDK default to hard coded values here but react native will pass it's own values here. DO NOT override this in your apps.
+public func overrideSDKDefaults(url: String? = nil, name: String? = nil, version: String? = nil) {
```

Comment:
would it be useful to show a warning to any callers of this function, telling them that it's for internal use only? If it would be useful, we could add something like
``` swift
@available(*, deprecated, message: "This function is for internal use only, and should not be used in production")
```
<img width="666" alt="Screenshot 2024-08-29 at 14 40 52" src="https://github.com/user-attachments/assets/1da57139-456f-4451-b9c0-addbbd56b642">

Reply:
Nice I like this. Adding it in now, we will see this on our test app but we can live with that.